crates/sui-rosetta/src/operations.rs

```diff
@@ -213,23 +249,33 @@ impl Operations {
         Ok(InternalOperation::WithdrawStake { sender, stake_ids })
     }
 
+    fn from_transaction_block_data(
+        data: SuiTransactionBlockData,
+        input_args: Option<Vec<CallArg>>,
+    ) -> Result<Operations, Error> {
+        let sender = *data.sender();
+        Self::from_transaction(data.transaction().clone(), sender, None, input_args)
+    }
+
     fn from_transaction(
         tx: SuiTransactionBlockKind,
         sender: SuiAddress,
         status: Option<OperationStatus>,
-    ) -> Result<Vec<Operation>, Error> {
+        input_args: Option<Vec<CallArg>>,
+    ) -> Result<Operations, Error> {
         Ok(match tx {
-            SuiTransactionBlockKind::ProgrammableTransaction(pt) => {
-                Self::parse_programmable_transaction(sender, status, pt)?
-            }
-            _ => vec![Operation::generic_op(status, sender, tx)],
+            SuiTransactionBlockKind::ProgrammableTransaction(pt) => Self::new(
+                Self::parse_programmable_transaction(sender, status, pt, input_args)?,
+            ),
+            _ => Self::new(vec![Operation::generic_op(status, sender, tx)]),
         })
     }
 
     fn parse_programmable_transaction(
         sender: SuiAddress,
         status: Option<OperationStatus>,
         pt: SuiProgrammableTransactionBlock,
+        input_args: Option<Vec<CallArg>>,
```

Comment:
does the `SuiProgrammableTransactionBlock` contains the call arg we need already?

Reply:
Hey @patrickkuo the reason for these `input_args` needed here is because while the `inputs` vector from the `SuiProgrammableTransactionBlock` contains the entry for this extra pure parameter I've added as the bearer of the currency info, it does not actually contain it's value. It seems that this data are not available in memory, I guess it has to do to the fact that this input is not effectively being used into any transaction (see attached img 1).
The workaround was to parse the raw tx block's data and deserialise the input objects so I can get the value in the extra `input_args` param you highlighted (see attached img 2). 
I' ll try to express this with a comment in the code as per your review comment above.

![Screenshot 2024-08-07 at 2 24 16 PM](https://github.com/user-attachments/assets/d80b35f6-19e9-4b69-9282-b6c814632c90)
![Screenshot 2024-08-07 at 2 25 06 PM](https://github.com/user-attachments/assets/093a8485-b0eb-4c97-bd5f-a16644651037)

Reply:
it's stored as JsonValue

Reply:
ya I tried locally and it works.

you can use this code to get the bytes from json value
```
         currency = pt.inputs.iter().last().and_then(|arg| {
                            if let SuiCallArg::Pure(value) = arg {
                                let bytes = value
                                    .value()
                                    .to_json_value()
                                    .as_array()?
                                    .clone()
                                    .into_iter()
                                    .map(|v| v.as_u64().map(|n| n as u8))
                                    .collect::<Option<Vec<u8>>>()?;
                                bcs::from_bytes::<String>(&bytes)
                                    .ok()
                                    .and_then(|bcs_str| serde_json::from_str(&bcs_str).ok())
                            } else {
                                None
                            }
                        });
```

Reply:
Awesome, thank you very much. I will amend accordingly then as it will simplify the flow.

Reply:
Done, thank you!